src/content/reference/gsctl/update-cluster.md

```diff
@@ -0,0 +1,56 @@
+---
+title: "gsctl Command Reference: update cluster"
+description: The 'gsctl update cluster' command allows the modification of the name of a cluster and its labels.
+date: 2020-05-04
+type: page
+weight: 44
+---
+
+# `gsctl update cluster`
+
+The `gsctl update cluster` allows the modification of the name of a cluster and its labels.
+Cluster labelling is only available for clusters with node pools on AWS.
+
+## Usage
+
+The command is called with the cluster ID or name as a positional argument.
+The desired new name can be specified with the `--name` or `-n` flag.
+The `--label` flag is used to modify labels.
+It can receive either a JSON formatted [JSON Merge Patch, RFC 7386](https://tools.ietf.org/html/rfc7386) or a string specifying label changes.
+
+### Name modification example
+
+```nohighlight
+$ gsctl update cluster f01r4 --name "Precious Production Cluster"
+```
+
+### Labels modification JSON Merge Patch example
+
+```nohighlight
+$ gsctl update cluster vxvc7 --labels '{"environment": "testing", "locked": null}'
+```
+
+will update the labels of cluster `vxvc7`. It will add (or update depending on prior existence) label `environment=testing` and delete the label with key `locked`.
+
+### Labels modification key=value example
+
+```nohighlight
+$ gsctl update cluster vxvc7 --labels 'environment=testing locked-'
```

Comment:
I would strongly suggest use of multiple `--label` flags here (not `--labels`). Also I don't like the custom "minus" syntax, an empty string should be more expressive.

```suggestion
$ gsctl update cluster vxvc7 --label environment=testing --label locked=
```

In spf13/cobra, using a `StringSlice` flag for the above means that the following syntax will also work:

```
$ gsctl update cluster vxvc7 --label environment=testing,locked=
```

Reply:
This is lifted from `kubectl labels` https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#label

Reply:
I would also like to align with kubectl here as this is also going to be the interaction basis in the future while we open our control plane.

Reply:
To me, aligning with a very specific kubectl behaviour in a tool that is in many ways different from kubectl is not a goal we should pursue.

That would be different if gsctl was a kubectl plugin already.